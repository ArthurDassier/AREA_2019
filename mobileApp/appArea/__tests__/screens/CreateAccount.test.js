// __tests__/Intro-test.js
import React from 'react';
import CreateAccount from '../../screens/CreateAccount';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';


describe('CreateAccount screen', () => {
  it('CreateAccount screen: renders correctly', () => {
    const tree = renderer.create(<CreateAccount />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('CreateAccount screen: Check initial state value', () => {
    const wrapper = shallow(<CreateAccount />);
    expect(wrapper.instance().state.showPwd).toBe(true);
    expect(wrapper.instance().state.pressed).toBe(false);
    expect(wrapper.instance().state.username).toBe("");
    expect(wrapper.instance().state.mail).toBe("");
    expect(wrapper.instance().state.password).toBe("");
    expect(wrapper.instance().state.confirmedPassword).toBe("");
    expect(wrapper.instance().state.emptyField).toBe(false);
    expect(wrapper.instance().state.wrongMail).toBe(false);
    expect(wrapper.instance().state.badPassword).toBe(false);
    expect(wrapper.instance().state.pwdDoesntMatch).toBe(false);
    expect(wrapper.instance().state.badAccount).toBe("");
  });
  it('CreateAccount screen: should put ON the secure password entry', () => {
    const wrapper = shallow(<CreateAccount />);
    const instance = wrapper.instance();
    instance._showPwd();
    expect(instance.state.showPwd).toBe(false);
    expect(instance.state.pressed).toBe(true);
    instance._showPwd();
    expect(instance.state.showPwd).toBe(true);
    expect(instance.state.pressed).toBe(false);
  });
  it('CreateAccount screen: check email validation format', () => {
    const wrapper = shallow(<CreateAccount />);
    const instance = wrapper.instance();
    expect(instance._validate("louis@gmail.com")).toBe(true);
    expect(instance._validate("louisgmail.com")).toBe(false);
    expect(instance._validate("louis@gmail.")).toBe(false);
    expect(instance._validate("@gmail.")).toBe(false);
  });
  it('CreateAccount screen: check account data', () => {
    const navigation = { navigate: jest.fn() };
    const wrapper = shallow(<CreateAccount navigation={navigation} />);
    const instance = wrapper.instance();
    instance._checkAccount({ "status": "success" })
    instance._checkAccount({ "message": "error" })
    const tree = renderer.create(wrapper).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('CreateAccount screen:  create an account', () => {
    const wrapper = shallow(<CreateAccount />);
    const instance = wrapper.instance();
    instance._createMyAccount();
    const emptyField = renderer.create(wrapper).toJSON();
    expect(emptyField).toMatchSnapshot();
    instance.setState({username: "a", password: "aaaaaa", confirmedPassword: "aaaaa", mail:"l"})
    instance._createMyAccount();
    const wrongMail = renderer.create(wrapper).toJSON();
    expect(wrongMail).toMatchSnapshot();
    instance.setState({mail: "louis@gmail.com"})
    instance._createMyAccount();
    const pwdDoesntMatch = renderer.create(wrapper).toJSON();
    expect(pwdDoesntMatch).toMatchSnapshot();
  });


});