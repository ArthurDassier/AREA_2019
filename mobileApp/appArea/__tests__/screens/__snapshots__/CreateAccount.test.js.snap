// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateAccount screen CreateAccount screen:  create an account 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "height": null,
      "justifyContent": "center",
      "width": null,
    }
  }
>
  <Image
    source={1}
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": null,
          "width": null,
        },
        undefined,
      ]
    }
  />
  <View
    onLayout={[Function]}
  >
    <View
      style={
        Object {
          "bottom": 0,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "marginBottom": 50,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 120,
              "width": 120,
            }
          }
        />
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 35,
              "fontWeight": "500",
              "opacity": 0.5,
            }
          }
        >
          AREA APP
        </Text>
      </View>
      <Text
        style={
          Object {
            "color": "rgba(255, 99, 71, 0.7)",
            "fontSize": 16,
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        Fields can't be empty
      </Text>
      <View
        style={
          Object {
            "marginTop": 10,
          }
        }
      >
        <Text />
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Username"
          placeholderTextColor="(rgba(255, 255, 255, 0.7)"
          rejectResponderTermination={true}
          style={
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0.35)",
              "borderRadius": 25,
              "color": "rgba(255, 255, 255, 0.7)",
              "fontSize": 16,
              "height": 45,
              "marginHorizontal": 25,
              "paddingLeft": 45,
              "width": 695,
            }
          }
          underlineColorAndroid="transparent"
        />
      </View>
      <View
        style={
          Object {
            "marginTop": 10,
          }
        }
      >
        <Text />
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Mail address"
          placeholderTextColor="(rgba(255, 255, 255, 0.7)"
          rejectResponderTermination={true}
          style={
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0.35)",
              "borderRadius": 25,
              "color": "rgba(255, 255, 255, 0.7)",
              "fontSize": 16,
              "height": 45,
              "marginHorizontal": 25,
              "paddingLeft": 45,
              "width": 695,
            }
          }
          underlineColorAndroid="transparent"
        />
      </View>
      <View
        style={
          Object {
            "marginTop": 10,
          }
        }
      >
        <Text />
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Password"
          placeholderTextColor="(rgba(255, 255, 255, 0.7)"
          rejectResponderTermination={true}
          secureTextEntry={true}
          style={
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0.35)",
              "borderRadius": 25,
              "color": "rgba(255, 255, 255, 0.7)",
              "fontSize": 16,
              "height": 45,
              "marginHorizontal": 25,
              "paddingLeft": 45,
              "width": 695,
            }
          }
          underlineColorAndroid="transparent"
        />
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
              "position": "absolute",
              "right": 37,
              "top": 8,
            }
          }
        >
          <Text />
        </View>
      </View>
      <View
        style={
          Object {
            "marginTop": 10,
          }
        }
      >
        <Text />
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Confirm your password"
          placeholderTextColor="(rgba(255, 255, 255, 0.7)"
          rejectResponderTermination={true}
          secureTextEntry={true}
          style={
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0.35)",
              "borderRadius": 25,
              "color": "rgba(255, 255, 255, 0.7)",
              "fontSize": 16,
              "height": 45,
              "marginHorizontal": 25,
              "paddingLeft": 45,
              "width": 695,
            }
          }
          underlineColorAndroid="transparent"
        />
      </View>
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#0B75D9",
        "borderRadius": 25,
        "height": 45,
        "justifyContent": "center",
        "marginTop": 20,
        "opacity": 1,
        "width": 695,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "rgba(255, 255, 255, 0.7)",
          "fontSize": 16,
          "textAlign": "center",
        }
      }
    >
      Create my account
    </Text>
  </View>
</View>
`;

exports[`CreateAccount screen CreateAccount screen:  create an account 2`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "height": null,
      "justifyContent": "center",
      "width": null,
    }
  }
>
  <Image
    source={1}
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": null,
          "width": null,
        },
        undefined,
      ]
    }
  />
  <View
    onLayout={[Function]}
  >
    <View
      style={
        Object {
          "bottom": 0,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "marginBottom": 50,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 120,
              "width": 120,
            }
          }
        />
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 35,
              "fontWeight": "500",
              "opacity": 0.5,
            }
          }
        >
          AREA APP
        </Text>
      </View>
      <Text
        style={
          Object {
            "color": "rgba(255, 99, 71, 0.7)",
            "fontSize": 16,
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        Wrong formatted email bitch
      </Text>
      <View
        style={
          Object {
            "marginTop": 10,
          }
        }
      >
        <Text />
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Username"
          placeholderTextColor="(rgba(255, 255, 255, 0.7)"
          rejectResponderTermination={true}
          style={
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0.35)",
              "borderRadius": 25,
              "color": "rgba(255, 255, 255, 0.7)",
              "fontSize": 16,
              "height": 45,
              "marginHorizontal": 25,
              "paddingLeft": 45,
              "width": 695,
            }
          }
          underlineColorAndroid="transparent"
        />
      </View>
      <View
        style={
          Object {
            "marginTop": 10,
          }
        }
      >
        <Text />
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Mail address"
          placeholderTextColor="(rgba(255, 255, 255, 0.7)"
          rejectResponderTermination={true}
          style={
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0.35)",
              "borderRadius": 25,
              "color": "rgba(255, 255, 255, 0.7)",
              "fontSize": 16,
              "height": 45,
              "marginHorizontal": 25,
              "paddingLeft": 45,
              "width": 695,
            }
          }
          underlineColorAndroid="transparent"
        />
      </View>
      <View
        style={
          Object {
            "marginTop": 10,
          }
        }
      >
        <Text />
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Password"
          placeholderTextColor="(rgba(255, 255, 255, 0.7)"
          rejectResponderTermination={true}
          secureTextEntry={true}
          style={
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0.35)",
              "borderRadius": 25,
              "color": "rgba(255, 255, 255, 0.7)",
              "fontSize": 16,
              "height": 45,
              "marginHorizontal": 25,
              "paddingLeft": 45,
              "width": 695,
            }
          }
          underlineColorAndroid="transparent"
        />
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
              "position": "absolute",
              "right": 37,
              "top": 8,
            }
          }
        >
          <Text />
        </View>
      </View>
      <View
        style={
          Object {
            "marginTop": 10,
          }
        }
      >
        <Text />
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Confirm your password"
          placeholderTextColor="(rgba(255, 255, 255, 0.7)"
          rejectResponderTermination={true}
          secureTextEntry={true}
          style={
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0.35)",
              "borderRadius": 25,
              "color": "rgba(255, 255, 255, 0.7)",
              "fontSize": 16,
              "height": 45,
              "marginHorizontal": 25,
              "paddingLeft": 45,
              "width": 695,
            }
          }
          underlineColorAndroid="transparent"
        />
      </View>
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#0B75D9",
        "borderRadius": 25,
        "height": 45,
        "justifyContent": "center",
        "marginTop": 20,
        "opacity": 1,
        "width": 695,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "rgba(255, 255, 255, 0.7)",
          "fontSize": 16,
          "textAlign": "center",
        }
      }
    >
      Create my account
    </Text>
  </View>
</View>
`;

exports[`CreateAccount screen CreateAccount screen:  create an account 3`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "height": null,
      "justifyContent": "center",
      "width": null,
    }
  }
>
  <Image
    source={1}
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": null,
          "width": null,
        },
        undefined,
      ]
    }
  />
  <View
    onLayout={[Function]}
  >
    <View
      style={
        Object {
          "bottom": 0,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "marginBottom": 50,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 120,
              "width": 120,
            }
          }
        />
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 35,
              "fontWeight": "500",
              "opacity": 0.5,
            }
          }
        >
          AREA APP
        </Text>
      </View>
      <View
        style={
          Object {
            "marginTop": 10,
          }
        }
      >
        <Text />
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Username"
          placeholderTextColor="(rgba(255, 255, 255, 0.7)"
          rejectResponderTermination={true}
          style={
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0.35)",
              "borderRadius": 25,
              "color": "rgba(255, 255, 255, 0.7)",
              "fontSize": 16,
              "height": 45,
              "marginHorizontal": 25,
              "paddingLeft": 45,
              "width": 695,
            }
          }
          underlineColorAndroid="transparent"
        />
      </View>
      <View
        style={
          Object {
            "marginTop": 10,
          }
        }
      >
        <Text />
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Mail address"
          placeholderTextColor="(rgba(255, 255, 255, 0.7)"
          rejectResponderTermination={true}
          style={
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0.35)",
              "borderRadius": 25,
              "color": "rgba(255, 255, 255, 0.7)",
              "fontSize": 16,
              "height": 45,
              "marginHorizontal": 25,
              "paddingLeft": 45,
              "width": 695,
            }
          }
          underlineColorAndroid="transparent"
        />
      </View>
      <View
        style={
          Object {
            "marginTop": 10,
          }
        }
      >
        <Text />
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Password"
          placeholderTextColor="(rgba(255, 255, 255, 0.7)"
          rejectResponderTermination={true}
          secureTextEntry={true}
          style={
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0.35)",
              "borderRadius": 25,
              "color": "rgba(255, 255, 255, 0.7)",
              "fontSize": 16,
              "height": 45,
              "marginHorizontal": 25,
              "paddingLeft": 45,
              "width": 695,
            }
          }
          underlineColorAndroid="transparent"
        />
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
              "position": "absolute",
              "right": 37,
              "top": 8,
            }
          }
        >
          <Text />
        </View>
      </View>
      <View
        style={
          Object {
            "marginTop": 10,
          }
        }
      >
        <Text />
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Confirm your password"
          placeholderTextColor="(rgba(255, 255, 255, 0.7)"
          rejectResponderTermination={true}
          secureTextEntry={true}
          style={
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0.35)",
              "borderRadius": 25,
              "color": "rgba(255, 255, 255, 0.7)",
              "fontSize": 16,
              "height": 45,
              "marginHorizontal": 25,
              "paddingLeft": 45,
              "width": 695,
            }
          }
          underlineColorAndroid="transparent"
        />
      </View>
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#0B75D9",
        "borderRadius": 25,
        "height": 45,
        "justifyContent": "center",
        "marginTop": 20,
        "opacity": 1,
        "width": 695,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "rgba(255, 255, 255, 0.7)",
          "fontSize": 16,
          "textAlign": "center",
        }
      }
    >
      Create my account
    </Text>
  </View>
</View>
`;

exports[`CreateAccount screen CreateAccount screen: check account data 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "height": null,
      "justifyContent": "center",
      "width": null,
    }
  }
>
  <Image
    source={1}
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": null,
          "width": null,
        },
        undefined,
      ]
    }
  />
  <View
    onLayout={[Function]}
  >
    <View
      style={
        Object {
          "bottom": 0,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "marginBottom": 50,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 120,
              "width": 120,
            }
          }
        />
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 35,
              "fontWeight": "500",
              "opacity": 0.5,
            }
          }
        >
          AREA APP
        </Text>
      </View>
      <Text
        style={
          Object {
            "color": "rgba(255, 99, 71, 0.7)",
            "fontSize": 16,
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        error
      </Text>
      <View
        style={
          Object {
            "marginTop": 10,
          }
        }
      >
        <Text />
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Username"
          placeholderTextColor="(rgba(255, 255, 255, 0.7)"
          rejectResponderTermination={true}
          style={
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0.35)",
              "borderRadius": 25,
              "color": "rgba(255, 255, 255, 0.7)",
              "fontSize": 16,
              "height": 45,
              "marginHorizontal": 25,
              "paddingLeft": 45,
              "width": 695,
            }
          }
          underlineColorAndroid="transparent"
        />
      </View>
      <View
        style={
          Object {
            "marginTop": 10,
          }
        }
      >
        <Text />
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Mail address"
          placeholderTextColor="(rgba(255, 255, 255, 0.7)"
          rejectResponderTermination={true}
          style={
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0.35)",
              "borderRadius": 25,
              "color": "rgba(255, 255, 255, 0.7)",
              "fontSize": 16,
              "height": 45,
              "marginHorizontal": 25,
              "paddingLeft": 45,
              "width": 695,
            }
          }
          underlineColorAndroid="transparent"
        />
      </View>
      <View
        style={
          Object {
            "marginTop": 10,
          }
        }
      >
        <Text />
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Password"
          placeholderTextColor="(rgba(255, 255, 255, 0.7)"
          rejectResponderTermination={true}
          secureTextEntry={true}
          style={
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0.35)",
              "borderRadius": 25,
              "color": "rgba(255, 255, 255, 0.7)",
              "fontSize": 16,
              "height": 45,
              "marginHorizontal": 25,
              "paddingLeft": 45,
              "width": 695,
            }
          }
          underlineColorAndroid="transparent"
        />
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
              "position": "absolute",
              "right": 37,
              "top": 8,
            }
          }
        >
          <Text />
        </View>
      </View>
      <View
        style={
          Object {
            "marginTop": 10,
          }
        }
      >
        <Text />
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Confirm your password"
          placeholderTextColor="(rgba(255, 255, 255, 0.7)"
          rejectResponderTermination={true}
          secureTextEntry={true}
          style={
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0.35)",
              "borderRadius": 25,
              "color": "rgba(255, 255, 255, 0.7)",
              "fontSize": 16,
              "height": 45,
              "marginHorizontal": 25,
              "paddingLeft": 45,
              "width": 695,
            }
          }
          underlineColorAndroid="transparent"
        />
      </View>
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#0B75D9",
        "borderRadius": 25,
        "height": 45,
        "justifyContent": "center",
        "marginTop": 20,
        "opacity": 1,
        "width": 695,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "rgba(255, 255, 255, 0.7)",
          "fontSize": 16,
          "textAlign": "center",
        }
      }
    >
      Create my account
    </Text>
  </View>
</View>
`;

exports[`CreateAccount screen CreateAccount screen: renders correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "height": null,
      "justifyContent": "center",
      "width": null,
    }
  }
>
  <Image
    source={1}
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": null,
          "width": null,
        },
        undefined,
      ]
    }
  />
  <View
    onLayout={[Function]}
  >
    <View
      style={
        Object {
          "bottom": 0,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "marginBottom": 50,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 120,
              "width": 120,
            }
          }
        />
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 35,
              "fontWeight": "500",
              "opacity": 0.5,
            }
          }
        >
          AREA APP
        </Text>
      </View>
      <View
        style={
          Object {
            "marginTop": 10,
          }
        }
      >
        <Text />
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Username"
          placeholderTextColor="(rgba(255, 255, 255, 0.7)"
          rejectResponderTermination={true}
          style={
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0.35)",
              "borderRadius": 25,
              "color": "rgba(255, 255, 255, 0.7)",
              "fontSize": 16,
              "height": 45,
              "marginHorizontal": 25,
              "paddingLeft": 45,
              "width": 695,
            }
          }
          underlineColorAndroid="transparent"
        />
      </View>
      <View
        style={
          Object {
            "marginTop": 10,
          }
        }
      >
        <Text />
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Mail address"
          placeholderTextColor="(rgba(255, 255, 255, 0.7)"
          rejectResponderTermination={true}
          style={
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0.35)",
              "borderRadius": 25,
              "color": "rgba(255, 255, 255, 0.7)",
              "fontSize": 16,
              "height": 45,
              "marginHorizontal": 25,
              "paddingLeft": 45,
              "width": 695,
            }
          }
          underlineColorAndroid="transparent"
        />
      </View>
      <View
        style={
          Object {
            "marginTop": 10,
          }
        }
      >
        <Text />
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Password"
          placeholderTextColor="(rgba(255, 255, 255, 0.7)"
          rejectResponderTermination={true}
          secureTextEntry={true}
          style={
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0.35)",
              "borderRadius": 25,
              "color": "rgba(255, 255, 255, 0.7)",
              "fontSize": 16,
              "height": 45,
              "marginHorizontal": 25,
              "paddingLeft": 45,
              "width": 695,
            }
          }
          underlineColorAndroid="transparent"
        />
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
              "position": "absolute",
              "right": 37,
              "top": 8,
            }
          }
        >
          <Text />
        </View>
      </View>
      <View
        style={
          Object {
            "marginTop": 10,
          }
        }
      >
        <Text />
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Confirm your password"
          placeholderTextColor="(rgba(255, 255, 255, 0.7)"
          rejectResponderTermination={true}
          secureTextEntry={true}
          style={
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0.35)",
              "borderRadius": 25,
              "color": "rgba(255, 255, 255, 0.7)",
              "fontSize": 16,
              "height": 45,
              "marginHorizontal": 25,
              "paddingLeft": 45,
              "width": 695,
            }
          }
          underlineColorAndroid="transparent"
        />
      </View>
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#0B75D9",
        "borderRadius": 25,
        "height": 45,
        "justifyContent": "center",
        "marginTop": 20,
        "opacity": 1,
        "width": 695,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "rgba(255, 255, 255, 0.7)",
          "fontSize": 16,
          "textAlign": "center",
        }
      }
    >
      Create my account
    </Text>
  </View>
</View>
`;
